src/heartbeat/utils.js

```diff
@@ -62,7 +58,20 @@ function changeCollectorStatus(currentStatus, newStatus) {
  */
 function updateCollectorConfig(cc) {
   const config = configModule.getConfig();
-  Object.keys(cc).forEach((key) => config.refocus[key] = cc[key]);
+  Object.keys(cc).forEach((key) => {
+    const oldValue = config.refocus[key];
+    const newValue = cc[key];
+    config.refocus[key] = cc[key];
+    if (oldValue && newValue !== oldValue) {
+      if (key === 'heartbeatIntervalMillis') {
+        repeater.updateHeartbeatRepeater(newValue);
+      } else if (key === 'sampleUpsertQueueTimeMillis') {
+        queue.updateFlushTimeoutAll(newValue);
+      } else if (key === 'maxSamplesPerBulkUpsert') {
+        queue.updateSizeAll(newValue);
+      }
+    }
+  });
```

Comment:
Need to update existing repeaters/queues when the config is updated in the heartbeat.